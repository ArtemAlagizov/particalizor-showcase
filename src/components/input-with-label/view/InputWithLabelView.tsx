import React from "react";
import { Styled } from "../styled";
import { IInputWithLabelView } from "./IInputWithLabelView";
import Input from "antd/es/input";

const InputWithLabelView: React.FC<IInputWithLabelView> = ({
  onChange,
  defaultValue,
  label
}) => {
  return (
    <Styled.InputWithLabelContainer>
      <Styled.InputLabel>{label}</Styled.InputLabel>
      <Styled.InputContainer>
        <Input
          placeholder={"set background color"}
          onChange={onChange}
          defaultValue={defaultValue}
        />
      </Styled.InputContainer>
    </Styled.InputWithLabelContainer>
  );
};

export default InputWithLabelView;
