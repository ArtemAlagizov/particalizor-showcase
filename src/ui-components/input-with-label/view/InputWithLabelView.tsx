import React from "react";
import { Styled } from "../styled";
import { IInputWithLabelView } from "./IInputWithLabelView";
import { Input } from "antd";

const InputWithLabelView: React.FC<IInputWithLabelView> = ({
  onChange,
  defaultValue,
  label
}) => {
  return (
    <Styled.InputWithLabelContainer>
      <Styled.InputLabel>{label}</Styled.InputLabel>
      <Styled.InputContainer>
        <Input
          placeholder={"set background color"}
          onChange={onChange}
          defaultValue={defaultValue}
        />
      </Styled.InputContainer>
    </Styled.InputWithLabelContainer>
  );
};

export default InputWithLabelView;
