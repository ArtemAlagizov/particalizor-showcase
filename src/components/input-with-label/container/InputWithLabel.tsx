import React from "react";
import InputWithLabelView from "../view/InputWithLabelView";
import { IInputWithLabel } from "./IInputWithLabel";

export const InputWithLabel: React.FC<IInputWithLabel> = ({
  propertyName,
  onValueChange,
  defaultValue,
  label
}) => {
  const onChange = (event: React.ChangeEvent<HTMLInputElement>) =>
    onValueChange(event.target.value, propertyName);

  return (
    <InputWithLabelView
      label={label}
      defaultValue={defaultValue}
      onChange={onChange}
    />
  );
};

export default InputWithLabel;
