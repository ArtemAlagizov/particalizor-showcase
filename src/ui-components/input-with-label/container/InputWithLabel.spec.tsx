import React from "react";
import { mount, shallow } from "enzyme";
import InputWithLabel from "./InputWithLabel";
import InputWithLabelView from "../view/InputWithLabelView";

describe("Given a InputWithLabel", () => {
  describe("when rendering", () => {
    it("should match snapshot", () => {
      const component = shallow(
        <InputWithLabel
          selectedValue={""}
          onValueChange={jest.fn()}
          label={""}
          defaultValue={""}
          propertyName={""}
        />
      );

      expect(component).toMatchSnapshot();
    });
    it("should render MovingPictureView", () => {
      const component = mount(
        <InputWithLabel
          selectedValue={""}
          onValueChange={jest.fn()}
          label={""}
          defaultValue={""}
          propertyName={""}
        />
      );

      expect(component.find(InputWithLabelView).length).toBe(1);
    });
  });
});
