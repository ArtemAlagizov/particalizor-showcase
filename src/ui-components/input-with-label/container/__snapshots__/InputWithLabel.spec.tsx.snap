// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a InputWithLabel when rendering should match snapshot 1`] = `
<InputWithLabelView
  defaultValue=""
  label=""
  onChange={[Function]}
/>
`;
